libraries/Microsoft.Bot.Builder.Alexa/Integration/AspNet.WebApi/AlexaRequestHandler.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Formatting;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Alexa.Helpers;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.Bot.Builder.Alexa.Integration.AspNet.WebApi
+{
+    internal sealed class AlexaRequestHandler : HttpMessageHandler
```

Comment:
>lexaRequestHandler : HttpMessageHandler [](start = 27, length = 39)

I guess I missed something.  You don't do this as a controller?  Have I been doing it wrong all of these years?

Reply:
You could use a controller here, but I am aiming for parity between the Bot Framework adapter and the integration layers we have for that which work in the same way.  This also means that someone using WebApi / .NET Core can add the Alexa adapter alongside the BF adapter in a consistent way for the same bot, which I think will be a common use case.

Reply:
@tomlm I thought I saw a comment last night about moving the deserialization into the adapter itself, but I don't see it now.  In any case I wanted to follow up.  This has again been done for parity with the existing integration layers in place for the BF adapter, where the deserialization of the incoming Activity happens within the integration layer before being passed to the Activity. I do kind of like the idea of having the Adapter take an Alexa specific model and return an Alexa specific response, but totally happy to discuss how this should work in the long term.  I do feel though that it is more important to maintain parity between the way the integration layers / adapters work for now as it would be great in the future to make the adapters more consistent and have a single generic integration layer (I believe we have had a DCR open to discuss adapter architecture started by @billba).